miner/worker.go

```diff
@@ -723,6 +723,15 @@ func (w *worker) commitTransactions(txs *types.TransactionsByPriceAndNonce, txFe
 		if tx == nil {
 			break
 		}
+		// Short-circuit if the transaction requires more gas than we have in the pool.
+		// If we didn't short-circuit here, we would get core.ErrGasLimitReached below.
+		// Short-circuiting here saves us the trouble of checking the GPM and so on when the tx can't be included
+		// anyway due to the block not having enough gas left.
+		if w.current.gasPool.Gas() < tx.Gas() {
+			log.Trace("Skipping transaction which requires more gas than is left in the block", "hash", tx.Hash(), "gas", w.current.gasPool.Gas(), "txgas", tx.Gas())
```

Comment:
i'd remove the trace

Reply:
I'm not too big a fan of it, but included it because of the corresponding trace below if you get `core.ErrGasLimitReached`.  Will keep it for now, and we can reconsider whether to remove some of these traces in the future.